scipy/stats/_distribution_infrastructure.py

```diff
@@ -0,0 +1,5082 @@
+            Tuple indicating the limits within which the quantities are plotted.
+            Default is ``('cdf', 0.001, 0.999)`` indicating that the central
+            99.9% of the distribution is to be shown. Valid values of the
+            string are the same as for ``x`` and ``y`` except for ``pdf``
+            and ``logpdf``.
+        ax : `matplotlib.axes`, optional
+            Axes on which to generate the plot. If not provided, use the
+            current axes.
+
+        Returns
+        -------
+        ax : `matplotlib.axes`, optional
+            Axes on which the plot was generated.
+            The plot can be customized by manipulating this object.
+
+        Examples
+        --------
+        Instantiate a distribution with the desired parameters:
+
+        >>> import numpy as np
+        >>> import matplotlib.pyplot as plt
+        >>> from scipy import stats
+        >>> X = stats.Normal(mu=1., sigma=2.)
+
+        Plot the PDF over the central 99.9% of the distribution.
+        Compare against a histogram of a random sample.
+
+        >>> ax = X.plot()
+        >>> sample = X.sample(10000, qmc_engine=stats.qmc.Halton)
+        >>> ax.hist(sample, density=True, bins=50, alpha=0.5)
+        >>> plt.show()
+
+        Plot ``logpdf(x)`` as a function of ``x`` in the left tail,
+        where the log of the CDF is between -10 and ``np.log(0.5)``.
+
+        >>> X.plot('x', 'logpdf', t=('logcdf', -10, np.log(0.5)))
+        >>> plt.show()
+
+        Plot the PDF of the normal distribution as a function of the
+        CDF for various values of the scale parameter.
+
+        >>> X = stats.Normal(mu=0., sigma=[0.5, 1., 2])
+        >>> X.plot('cdf', 'pdf')
+        >>> plt.show()
+
+        """
+
+        # Strategy: given t limits, get quantile limits. Form grid of
+        # quantiles, compute requested x and y at quantiles, and plot.
+        # Currently, the grid of quantiles is always linearly spaced.
+        # Instead of always computing linearly-spaced quantiles, it
+        # would be better to choose:
+        # a) quantiles or probabilities
+        # b) linearly or logarithmically spaced
+        # based on the specified `t`.
+        # TODO:
+        # - smart spacing of points
+        # - when the parameters of the distribution are an array,
+        #   use the full range of abscissae for all curves
+
+        t_is_quantile = {'x', 'icdf', 'iccdf', 'ilogcdf', 'ilogccdf'}
+        t_is_probability = {'cdf', 'ccdf', 'logcdf', 'logccdf'}
+        valid_t = t_is_quantile.union(t_is_probability)
+        valid_xy =  valid_t.union({'pdf', 'logpdf'})
+
+        ndim = len(self._shape)
+        x_name, y_name = x, y
+        t_name, tlim = t[0], np.asarray(t[1:])
+        tlim = tlim[:, np.newaxis] if ndim else tlim
+
+        # pdf/logpdf are not valid for `t` because we can't easily invert them
+        message = (f'Argument `t` of `{self.__class__.__name__}.plot` "'
```

Comment:
> did you consider having a name field instead

After writing this all over the place, yes.

> Also, a matter of style as not that consuming, but you compute these a bunch of time + these could be inside the conditions.

?

Reply:
I mean that `message` could be under the `if`, not before as it's not relevant if the condition is false.

Reply:
TBH, I usually do this to reduce the indentation I'm starting at when writing the message to avoid line breaks, and because the message serves as documentation for what the `if` block is for.